### ISSUE TITLE
[Recommender] Incorrect testing case generated for unary method map parameters in testing

### ISSUE THREAD

**Issue by viacheslav-rostovtsev** [googleapis/gapic-generator-ruby#332]
see also: https://github.com/googleapis/gapic-generator-ruby/issues/326
---
TL/DR: when we generate default empty map for an unary method we also generate a test case to see if it's first element is of a type of that's map's entry. Since the first element is nil the test case fails  
---
When generating Recommender v1beta1 with synth tool (see [Recommender PR 4734 in google-cloud-ruby](https://github.com/googleapis/google-cloud-ruby/pull/4734)):

if we run tests with 
```sh
bundle exec rake ci
```
we get an error like this for some testing methods:
```
Error:
Google::Cloud::Recommender::V1::Recommender::ClientTest#test_mark_recommendation_claimed:
NameError: uninitialized constant Google::Cloud::Recommender::V1::MarkRecommendationClaimedRequest::StateMetadataEntry
    /usr/local/google/home/virost/src/google-cloud-ruby/google-cloud-recommender-v1/test/google/cloud/recommender/v1/recommender_test.rb:191:in `block in test_mark_recommendation_claimed'
    /usr/local/google/home/virost/src/google-cloud-ruby/google-cloud-recommender-v1/test/google/cloud/recommender/v1/recommender_test.rb:40:in `call_rpc'
    /usr/local/google/home/virost/src/google-cloud-ruby/google-cloud-recommender-v1/lib/google/cloud/recommender/v1/recommender/client.rb:336:in `mark_recommendation_claimed'
    /usr/local/google/home/virost/src/google-cloud-ruby/google-cloud-recommender-v1/test/google/cloud/recommender/v1/recommender_test.rb:203:in `block in test_mark_recommendation_claimed'

```
in the recommender_test we see:
```ruby
    # Create request parameters for a unary method.
    state_metadata = {}
    (...)
    assert_kind_of Google::Cloud::Recommender::V1::MarkRecommendationClaimedRequest::StateMetadataEntry, request.state_metadata.first
```
where `request.state_metadata` is initialized with `state_metadata`:

```ruby
# Use protobuf object
      client.mark_recommendation_claimed Google::Cloud::Recommender::V1::MarkRecommendationClaimedRequest.new(name: name, state_metadata: state_metadata, etag: etag) do |response, operation|
        assert_equal grpc_response, response
        assert_equal grpc_operation, operation
      end
```

Since the `.first` of an empty map is nil, the `assert_kind_of` fails.

**Comment by dazuma** [comment]
Yeah that test seems incorrect. It probably needs to detect that the field is a map type (similar to how we did so when generating empty data) and respond accordingly. Proto doesn't return a hash when getting the field, but has its own map type that we should check against.
